meteor/client/lib/EditAttribute.tsx

```diff
@@ -913,6 +919,121 @@ const EditAttributeJson = wrapEditAttribute(
 		}
 	}
 )
+const EditAttributeArray = wrapEditAttribute(
+	class EditAttributeArray extends EditAttributeBase {
+		constructor(props) {
+			super(props)
+
+			this.handleChange = this.handleChange.bind(this)
+			this.handleBlur = this.handleBlur.bind(this)
+			this.handleEscape = this.handleEscape.bind(this)
+		}
+		isArray(strOrg: string): { parsed: any[] } | false {
+			if (!(strOrg + '').trim().length) return { parsed: [] }
+
+			const values: any[] = []
+			const strs = (strOrg + '').split(',')
+
+			for (const str of strs) {
+				// Check that the values in the array are of the right type:
+
+				if (this.props.arrayType === 'boolean') {
+					const parsed = JSON.parse(str)
+					if (typeof parsed !== 'boolean') return false // type check failed
+					values.push(parsed)
+				} else if (this.props.arrayType === 'int') {
+					const parsed = parseInt(str, 10)
+
+					if (Number.isNaN(parsed)) return false // type check failed
+					values.push(parsed)
+				} else if (this.props.arrayType === 'float') {
+					const parsed = parseFloat(str)
+					if (Number.isNaN(parsed)) return false // type check failed
+					values.push(parsed)
+				} else {
+					// if (this.props.arrayType === 'string')
+					let parsed: string
+					try {
+						parsed = JSON.parse(str)
+					} catch (_err) {
+						// ignore error
+						parsed = str + ''
+					}
+					if (typeof parsed !== 'string') return false // type check failed
```

Comment:
Also, strings that will look like a JSON object - but perhaps aren't, and need not be treated as such - will fail? Also, something like "1,2,3,4,MP1,MP2" will fail here, and I strongly think that it shouldn't ("1" is a perfectly acceptable string label).

Reply:
This should now be okay after having removed the JSON.parse part.